transiflow/interface/PETSc.py

```diff
@@ -77,21 +91,19 @@ def vector(self):
     def vector_from_array(self, array):
         return Vector.from_array(self.map, array[self.map_natural.indices], ghosts=self.ghosts)
 
-    def ghosts_petsc(self, ghosts_natural):
-        """Get global indices of ghosts in PETSc ordering.
-
-        1. On each process, create a vector containing the owned natural indices.
-        2. Gather the indices from all processes in a sequential vector and find the
-           locations of the ghosts in the sequential vector. The locations correspond to
-           the ghost indices in PETSc ordering.
-        """
-        vec_indices = Vector().createWithArray(self.map_natural.indices)
-        sct, vec_global_indices = PETSc.Scatter().toAll(vec_indices)
-        sct.scatter(vec_indices, vec_global_indices)
+    def array_from_vector(self, vector):
+        array = Vector.all_gather(vector).array
+        array_natural = array.copy()
+        array_natural[self.indices_natural_global_vec.array.astype(PETSc.IntType)] = array[:]
```

Comment:
I thought there was some numpy function that did this, but I also can't find it.

Reply:
do you mean `numpy.argsort`?

Reply:
That could be the one I was thinking off, yeah.

Reply:
yes, that's what I used for the other permutation vector. looks better and more consistent

Reply:
That looks great! Can you squash the first and last commit? Then I'll merge it.

Reply:
sure, done